packages/walletconnect/src/provider.ts

```diff
@@ -113,26 +113,47 @@ export class WalletConnectProvider extends SignerProvider {
   }
 
   public async connect(): Promise<void> {
-    const { uri, approval } = await this.client.connect({
-      requiredNamespaces: {
-        alephium: {
-          chains: [this.permittedChain],
-          methods: this.methods,
-          events: ['accountChanged']
+    if (!this.session) {
+      const { uri, approval } = await this.client.connect({
+        requiredNamespaces: {
+          alephium: {
+            chains: [this.permittedChain],
+            methods: this.methods,
+            events: ['accountChanged']
+          }
         }
+      })
+
+      if (uri) {
+        this.emitEvents('displayUri', uri)
       }
-    })
 
-    if (uri) {
-      this.emitEvents('displayUri', uri)
+      this.session = await approval()
+      this.updateNamespace(this.session.namespaces)
+    } else {
+      this.updateNamespace(this.session.namespaces)
+      await this.ping(this.session.topic)
```

Comment:
When the dApp auto connects but the wallet is not open, sending requests will be blocked. Since walletconnect allows a timeout range of 5 minutes to 7 days, we need to set an appropriate timeout ourselves.

Reply:
We should add a timeout to extension wallet connection as well, so the wallet will not auto-connect forever